compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/irCodegenUtils.kt

```diff
@@ -559,3 +568,18 @@ fun getLabeledThisName(callableName: String, prefix: String, defaultName: String
         defaultName
     } else prefix + mangleNameIfNeeded(callableName)
 }
+
+val IrAnnotationContainer.deprecationFlags: Int
+    get() {
+        val annotation = annotations.findAnnotation(FQ_NAMES.deprecated) ?: return 0
+        val isHidden = (annotation.getValueArgument(2) as? IrGetEnumValue)?.symbol?.owner?.name?.asString() == "HIDDEN"
+        return Opcodes.ACC_DEPRECATED or if (isHidden) Opcodes.ACC_SYNTHETIC else 0
+    }
+
+val IrFunction.deprecationFlags: Int
+    get() {
+        val isPropertyAnnotationStub = origin == JvmLoweredDeclarationOrigin.SYNTHETIC_METHOD_FOR_PROPERTY_ANNOTATIONS
```

Comment:
Unfortunately this doesn't work for `$annotations` methods that are moved to `DefaultImpls` classes for properties declared in interfaces because their origin is rewritten to `DEFAULT_IMPLS`: https://github.com/JetBrains/kotlin/blob/0065236bdebefa72c06aa7b32fe78bec6eb65781/compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ClassCodegen.kt#L275

Consider extracting that check and the comment and reusing it here, until we find a solution to this problem

Reply:
That's irritating, but thanks for the pointer. I've added the same workaround for now.